ngraph_creator/operations/src/Space_To_Batch.cpp

```diff
@@ -0,0 +1,124 @@
+//#define LOG_NDEBUG 0
+#include <Space_To_Batch.hpp>
+#define LOG_TAG "Space_To_Batch"
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+Space_To_Batch::Space_To_Batch(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool Space_To_Batch::validate() {
+    if (!checkInputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkInputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM)) {
+        return false;
+    }
+    if (!checkOutputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkOutputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM)) {
+        return false;
+    }
+    if (!checkInputOperandType(1, (int32_t)OperandType::TENSOR_INT32)) {
+        return false;
+    }
+    if (!checkInputOperandType(2, (int32_t)OperandType::TENSOR_INT32)) {
+        return false;
+    }
+
+    const auto& inputsSize = sModelInfo->getOperationInputsSize(mNnapiOperationIndex);
+    if (inputsSize == 4) {
+        if (!checkInputOperandType(3, (int32_t)OperandType::BOOL)) {
+            return false;
+        }
+    }
+
+    // Check input rank
+    const auto inputRank = getInputOperandDimensions(0).size();
+
+    if (inputRank > 4) return false;
+
+    if (inputRank <= 3) return false;
+
+    auto& block_shape_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 1);
+    // TODO: Add Support for all_tensors_as_inputs
+    if (!sModelInfo->isOperandLifeTimeConst(block_shape_OperandIndex)) {
+        ALOGE("%s Only Constant dimensions supported now", __func__);
+        return false;
+    }
+
+    auto crop_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 2);
+    // TODO: Add Support for all_tensors_as_inputs
+    if (!sModelInfo->isOperandLifeTimeConst(crop_OperandIndex)) {
+        ALOGE("%s Only Constant dimensions supported now", __func__);
+        return false;
+    }
+
+    ALOGV("%s PASSED", __func__);
+    return true;
+}
+
+std::shared_ptr<ngraph::Node> Space_To_Batch::createNode() {
+    int32_t layout = 0;
+    bool useNchw = false;
+    const auto& inputsSize = sModelInfo->getOperationInputsSize(mNnapiOperationIndex);
+
+    auto inputNode = getInputNode(0);
+
+    auto& inDims = getInputOperandDimensions(0);
+
+    const auto& block_shape_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 1);
+    auto block_shape = sModelInfo->GetConstVecOperand<int32_t>(block_shape_OperandIndex);
+
+    // Compensation for the shape to be same as the size of data input shape
+    block_shape.insert(block_shape.begin(), 1);
+    block_shape.insert(block_shape.begin(), 1);
+
+    const auto& crop_OperandIndex = sModelInfo->getOperationInput(mNnapiOperationIndex, 2);
+    // Fetch the 2D crop as a 1D vector, and then split it into 2
+    auto crop_2d = sModelInfo->GetConstVecOperand<int32_t>(crop_OperandIndex);
```

Comment:
Why are we using the name crop here ? Aren't these paddings ?

Reply:
Crop is the name being used in the openvino ngraph operation usage. So named it based on that 
Reference : https://docs.openvinotoolkit.org/latest/openvino_docs_ops_movement_BatchToSpace_2.html
Let me know if this needs to be changed

Reply:
For space to batch, the names are paddings*:
https://docs.openvinotoolkit.org/latest/openvino_docs_ops_movement_SpaceToBatch_2.html